feature-toggles/test/specs/auth/authServiceSpec.js

```diff
@@ -0,0 +1,124 @@
+        });
+    });
+
+    describe('AuthService user informations', function() {
+        beforeEach(function() {
+            authService.setupUser(userTest.accessToken);
+        });
+
+        it('should authService.getCurrentUser()', function() {
+            spyOn(userService, 'load').and.callThrough();
+            authService.setupUser(userTest.accessToken, userTest.emailVerified);
+            var user = authService.getCurrentUser();
+            var new_user = new userFactory.user(userTest);
+            expect(user).toEqual(new_user);
+        });
+
+        it('should authService.getUserToken()', function(done) {
+            spyOn(authService, 'save');
+            authService.getUserToken().then(userToken => {
+                expect(userToken).toEqual(firebaseUser.accessToken);
+                expect(authService.save).toHaveBeenCalled();
+                done();
+            });
+        });
+
+        it('should authService.isLoggedIn()', function() {
+            var isLoggedIn = authService.isLoggedIn();
+            expect(isLoggedIn).toEqual(true);
+        });
+
+        it('should authService.save()', function() {
+            spyOn(userService, 'load').and.callThrough();
+            authService.setupUser(userTest.accessToken, userTest.emailVerified);
+
+            window.localStorage.userInfo = null;
+            authService.save();
+            var userCache = window.localStorage.userInfo;
+            var new_user = JSON.stringify(userTest);
+
+            expect(userCache).toEqual(new_user);
+        });
+    });
+
+    describe('_getIdToken', function() {
+        beforeEach(function() {
+            authService.setupUser(userTest.accessToken);
+            authService.resolveTokenPromise = () => {};
+            spyOn(authService, 'resolveTokenPromise').and.callFake(() => {});
+        });
+
+        it('should refresh token when the request of new token is successful', function(done) {
+            const user = {
+                accessToken: "riuewyirouyweiuryiu21y3iuyiuwyeiudsjikahkjsah",
+                getIdToken: async () => user.accessToken
+            };
+            const savedResolveTokenPromisse = authService.resolveTokenPromise;
+
+            authService._getIdToken(user).then(function(accessToken) {
+                expect(accessToken).toEqual(user.accessToken);
+                expect(savedResolveTokenPromisse).toHaveBeenCalled();
+                done();
+            });
+        });
+
+        it('should refresh token when the request of new token fail', function(done) {
+            const user = {
+                accessToken: "riuewyirouyweiuryiu21y3iuyiuwyeiudsjikahkjsah",
+                getIdToken: async () => {throw "Network error!"}
+            };
+            const savedResolveTokenPromisse = authService.resolveTokenPromise;
+
+            authService._getIdToken(user).then(function(accessToken) {
+                expect(accessToken).toEqual(firebaseUser.accessToken);
```

Comment:
it was not clear, why it should be equal to firebase access token. Also the two description of this test is almost equals. should refresh when it fails or not. shouldnt be something different?